core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2570,10 +2570,23 @@ class SparkContext(config: SparkConf) extends Logging {
 
   private[spark] def newShuffleId(): Int = nextShuffleId.getAndIncrement()
 
-  private val nextRddId = new AtomicInteger(0)
+  private var nextRddId = new AtomicInteger(0)
```

Comment:
What happens when the RDD ID overflows within a SparkContext? Are there tests that cover these cases?

Curious if it would be better to switch to an AtomicLong and just modulo max int?

```suggestion
  private[spark] def newRddId(): Int = nextRddId.getAndIncrement() % Integer.MAX_VALUE.toInt
```

Reply:
What happens when the RDD ID overflows within a SparkContext? 
When BlockManager generates BlockId, BlockId only supports positive rddid, so BlockId generation fails.

switch to an AtomicLong ?
The scope of influence is very broad, which may require extensive discussion.